mantidproject/mantid
# Fix various issues with documentation build

**Description of work.**

#29733 noted various issues with the doc tests + html build when building without MantidPlot. This corrects these issues along with dealing with scenarios where a developer has a non-default facility/instrument combination - the doc tests should now be robust against this. 

**To test:**

* Checkout and build both the `docs-html` & `docs-doctest` targets.
* All tests should pass and there should be no warnings on the build.

Fixes #29733



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
